src/library/cratefeature.cpp

```diff
@@ -128,17 +143,71 @@ int CrateFeature::crateIdFromIndex(QModelIndex index) {
 
     QString dataPath = item->dataPath().toString();
     bool ok = false;
-    int playlistId = dataPath.toInt(&ok);
+    int crateId = dataPath.toInt(&ok);
     if (!ok) {
         return -1;
     }
-    return playlistId;
+    return crateId;
+}
+
+QModelIndex CrateFeature::indexFromCrateId(int id) {
+
+    QStack<QModelIndex> cratesToProcess;
+    QModelIndex crate;
+    cratesToProcess.push(QModelIndex());
+    while (! cratesToProcess.empty()) {
+        crate = cratesToProcess.pop();
+
+        if (id == crateIdFromIndex(crate)) {
+            qDebug() << "Found index of crate found id: " << id;
+            return crate;
+        }
+
+        for(int row = 0; row < m_childModel.rowCount(crate); row++) {
+            cratesToProcess.push(m_childModel.index(row, 0, crate));
```

Comment:
This can be improved by using a recursive function. This helps to get rid of the "cratesToProcess" stack.